lib/lightspeed/client.rb

```diff
@@ -26,25 +29,38 @@ def load_json(json)
     end
 
     def get(**args)
-      request(args.merge(method: :get)).perform
+      perform_request(:get, 1, **args)
     end
 
     def post(**args)
-      request(args.merge(method: :post)).perform
+      perform_request(:post, 10, **args)
     end
 
     def put(**args)
-      request(args.merge(method: :put)).perform
+      perform_request(:put, 10, **args)
     end
 
     def delete(**args)
-      request(args.merge(method: :delete)).perform
+      perform_request(:delete, 10, **args)
     end
 
     private
 
+    def perform_request(method, units, **args)
```

Comment:
I'd prefer all this was in the Lightspeed::Request class, that you tell the request to do the thing, and it decides whether it should or not. I realise that makes class instance variables probably necessary.